xarray/backends/h5netcdf_.py

```diff
@@ -55,43 +86,63 @@ def __init__(self, filename, mode='r', format=None, group=None,
         opener = functools.partial(_open_h5netcdf_group, filename, mode=mode,
                                    group=group)
         self.ds = opener()
+        self._isopen = True
         self.format = format
         self._opener = opener
         self._filename = filename
         self._mode = mode
         super(H5NetCDFStore, self).__init__(writer)
 
+    @contextlib.contextmanager
+    def ensure_open(self, autoclose=True):
```

Comment:
Also, why does `autoclose` default to True? This will start auto-opening/closing Datasets even if they were explicitly closed by calling `.close()`. That seems undesirable.

Possibly this should be saved as another DataStore attribute.